agenta-backend/agenta_backend/services/db_manager.py

```diff
@@ -607,26 +635,48 @@ async def get_user_object(user_uid: str) -> UserDB:
 
     user = await engine.find_one(UserDB, UserDB.uid == user_uid)
     if user is None:
-        org = OrganizationDB()
-        return UserDB(uid="0", organization_id=org)
-    return user
+        create_user = UserDB(uid="0")
+        await engine.save(create_user)
 
+        org = OrganizationDB(owner=create_user)
+        await engine.save(org)
 
-async def get_user_organization(user_id: str) -> OrganizationDB:
-    """Get the user organization object from the database.
+        create_user.organizations.append(org.id)
+        await engine.save(create_user)
+        await engine.save(org)
+
+        return create_user
+    else:
+        return user
+
+
+async def get_user_organizations(
+    user_id_or_user: Union[str, UserDB] = None
+) -> List[OrganizationDB]:
+    """Get the list of the user's organizations from the database.
 
     Arguments:
-        user_id (str): The user unique identifier
+        user_id_or_user (Union[str, UserDB]): Either the user's unique identifier (str) or the user object (UserDB).
 
     Returns:
-        OrganizationDB: instance of user organization
+        List[OrganizationDB]: list of user's organizations.
     """
+    print("This function was called.")
 
-    user = await get_user_object(user_id)
-    organization = await engine.find_one(
-        OrganizationDB, OrganizationDB.id == user.organization_id
-    )
-    return organization
+    if isinstance(user_id_or_user, str):
+        user = await get_user_object(user_id_or_user)
+    elif isinstance(user_id_or_user, UserDB):
+        user = user_id_or_user
+    else:
+        raise ValueError("user_id_or_user must be either a string or UserDB object")
+
+    organizations = []
+
+    if user.organizations is not None:
```

Comment:
what is the type of user.organizations? would list(user.organizations) not work?

Reply:
why would user.organizations is None. In that case wouldn't an exception be better?

if user.organizations is None:
raise Exception
else:
return list(user.organizations)

Reply:
> what is the type of user.organizations? would list(user.organizations) not work?

user.organizations is already a list.

Reply:
> why would user.organizations is None. In that case wouldn't an exception be better?
> 
> if user.organizations is None: raise Exception else: return list(user.organizations)

In the case that user.organizations is none which shouldn't be since a user should have a default either organization, we should either create one at that point or raise an exception. 

Which do you think works better?
@mmabrouk